giskard/llm/utils/validate_test_case.py

```diff
@@ -0,0 +1,195 @@
+    from langchain.chains.openai_functions import create_structured_output_chain
+except ImportError as err:
+    raise LLMImportError() from err
+
+
+class TestResult(PydanticBaseModel):
+    score: int = Field(
+        description="A number ranging from 1 to 5: 1 indicates that the answer does not meet the criteria at all, 3 indicates that the answer can be improved, 5 indicates that the answer completely meets the criteria"
+    )
+    reason: str = Field(description="A text that clearly explains the given score")
+    tip: Optional[str] = Field(
+        description="A text that offers a clear and descriptive suggestion on how to enhance the model"
+    )
+
+
+prompt = ChatPromptTemplate.from_messages(
+    [
+        (
+            "system",
+            """
+You are a prompt QA responsible for evaluating the LLM answer based on test assertions of the {model_name} model.
+
+**Scope**: Please focus on testing '{test_case}'.
+
+**Description**: The objective is to verify that the model answer related to the user prompt passes the test correctly.
+
+**Model Information**:
+``` 
+{model_description}
+```
+
+**Format Instructions**: 
+Please provide a JSON object as a response, including the following keys:
+- Score: A number ranging from 1 to 5:
+  - 1 indicates that the answer does not meet the criteria at all
+  - 3 indicates that the answer can be improved
+  - 5 indicates that the answer completely meets the criteria
+- Reason: A text that clearly explains the given score
+- Tip: A text that offers a clear and descriptive suggestion on how to enhance the model
+
+Example: 
+{{"score": 1, "reason": "The answer is missing the required key 'tips' in the JSON response", "tip": "To ensure that the schema is respected, provide an example of the expected JSON response"}}
+""",
+        ),
+        (
+            "user",
+            """
+```json
+{prompt}
+```
+
+You are auditing an AI model. Your task is to evaluate the model in the context of the following issue category
+
+Issue category: {issue_category_name}
+Issue category description: {issue_category_description}
+
+You are auding the following model:
+
+Model name: {model_name}
+Model description: {model_description}
+
+Here is the input that was provided to the model and its output:
+
+PROVIDED INPUT:
+###
+{input_vars}
+###
+
+MODEL OUTPUT:
+###
+{model_output}
+###
+
+Based on the information above, please check that the output satisfies all of the following requirements:
+{requirements}
+
+If the model satisfies the requirements, it passes the evaluation test. On the contrary, if the model output does not satisfy the requirements, the model fails the test.
+Call the `evaluate_model` function with the result of your evaluation.
+If the model does not pass the test, also provide a brief reason as an argument to the `evaluate_model`.
+If you are not sure, just answer 'I don’t know'.
+""",
+        ),
+        ("ai", "{response}"),
+    ]
+)
+
+chain = create_structured_output_chain(
```

Comment:
I guess we can delete `prompt` and `chain` variables since they are unused now

Reply:
I’ve not checked the tests yet, I thought it would break if I remove this. I’m testing the server integration today